diff --git a/third_party/inputmask/inputmask.date.extensions.js b/third_party/inputmask/inputmask.date.extensions.js
index b6e823315..d7a6ed743 100755
--- a/third_party/inputmask/inputmask.date.extensions.js
+++ b/third_party/inputmask/inputmask.date.extensions.js
@@ -252,6 +252,24 @@ export function factory(Inputmask) {
                     input.inputmask._valueSet(date);
                     $(input).trigger("setvalue");
                 }
+
+                // If the user presses backspace in the middle of the input
+                // value, stop masking until the field is cleared.
+                if (e.keyCode === Inputmask.keyCode.BACKSPACE) {
+                    const length = input.value.length;
+                    const cursorIsInMiddle = input.selectionStart != length &&
+                        input.selectionEnd != length;
+                    if (length > 0 && cursorIsInMiddle) {
+                        const cachedController = input.inputmask;
+                        input.inputmask.remove();
+                        input.addEventListener('input', function onclear() {
+                            if (input.value.length == 0) {
+                                input.removeEventListener('input', onclear);
+                                cachedController.mask(input);
+                            }
+                        });
+                    }
+                }
             },
             onUnMask: function(maskedValue, unmaskedValue, opts) {
                 return parse(opts.outputFormat, analyseMask(maskedValue, opts.inputFormat, opts), opts, true);
