--- a/third_party/caja/html-sanitizer.js
+++ b/third_party/caja/html-sanitizer.js
@@ -1018,6 +1018,7 @@ html4.ATTRIBS = {
   'td::scope': 0,
   'td::valign': 0,
   'td::width': 0,
+  'template::type': 0,
   'textarea::accesskey': 0,
   'textarea::autocomplete': 0,
   'textarea::cols': 0,
@@ -1197,6 +1198,7 @@ html4.ELEMENTS = {
   'table': 0,
   'tbody': 1,
   'td': 1,
+  'template': 4,
   'textarea': 8,
   'tfoot': 1,
   'th': 1,
@@ -1321,6 +1323,7 @@ html4.ELEMENT_DOM_INTERFACES = {
   'table': 'HTMLTableElement',
   'tbody': 'HTMLTableSectionElement',
   'td': 'HTMLTableDataCellElement',
+  'template': 'HTMLTemplateElement',
   'textarea': 'HTMLTextAreaElement',
   'tfoot': 'HTMLTableSectionElement',
   'th': 'HTMLTableHeaderCellElement',
